include/lbann/utils/dnn_lib/miopen/upsample.hpp

```diff
@@ -0,0 +1,115 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2014-2024, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory.
+// Written by the LBANN Research Team (B. Van Essen, et al.) listed in
+// the CONTRIBUTORS file. <lbann-dev@llnl.gov>
+//
+// LLNL-CODE-697807.
+// All rights reserved.
+//
+// This file is part of LBANN: Livermore Big Artificial Neural Network
+// Toolkit. For details, see http://software.llnl.gov/LBANN or
+// https://github.com/LLNL/LBANN.
+//
+// Licensed under the Apache License, Version 2.0 (the "Licensee"); you
+// may not use this file except in compliance with the License.  You may
+// obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the license.
+////////////////////////////////////////////////////////////////////////////////
+#ifndef LBANN_UTILS_DNN_LIB_MIOPEN_UPSAMPLE_HPP_
+#define LBANN_UTILS_DNN_LIB_MIOPEN_UPSAMPLE_HPP_
+
+#include "lbann/utils/dnn_enums.hpp"
+#include "lbann/utils/dnn_lib/helpers.hpp"
+#include "lbann/utils/gpu/helpers.hpp"
+
+#include "utils.hpp"
+
+namespace lbann {
+
+#ifdef LBANN_HAS_MIOPEN
+namespace dnn_lib {
+
+using namespace miopen;
+
+template <typename TensorDataType, typename ScalarParameterType>
+void upsample_nearest_forward(PoolingDescriptor const& poolingDesc,
+                              ScalarParameterType const& alpha_in,
+                              TensorDescriptor const& xDesc,
+                              TensorDataType const* x,
+                              ScalarParameterType const& beta_in,
+                              TensorDescriptor const& yDesc,
+                              TensorDataType* y,
+                              dnnHandle_t handle)
+{
+  LBANN_ERROR("Upsample layer is not currently supported on ROCm devices.");
```

Comment:
Why is that? Does MIOpen not have pooling?

Reply:
MIOpen has the extra complication of requiring a workspace and BVE suggested coming back to this later.